hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/EmptyWritableStates.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.spi.state;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+
+/** An implementation of {@link WritableStates} that is always empty. */
+public class EmptyWritableStates implements WritableStates {
+    private static final Set<String> EMPTY_SET = Collections.emptySet();
```

Comment:
Do you do this to have 1 step less in the stack? Or do I miss anything?

Reply:
Ah! I just assumed emptySet() created a new one every time, but I see now that it returns the same instance. So I can just return emptySet() (unless this turns out to be a performance problem, which I doubt, as I assume hotspot will inline it, and even if it doesn't, this method is probably not called frequently).